diff --git a/src/common.c b/src/common.c
index c2e1189..0e9ac17 100644
--- a/src/common.c
+++ b/src/common.c
@@ -33,7 +33,15 @@ copy_from_unicode(
	uint32 i;

	for (i = 0; i < len; i++) {
-		pdst[i] = psrc[i] & 0xFF;
+                //endiness check
+                if (psrc[i] >> 8 == 0x00)
+                {
+                  pdst[i] = psrc[i] & 0xFF;
+                }
+                else
+                {
+                  pdst[i] = (psrc[i] >> 8) & 0xFF;
+                }
	}
	return i;
 }
