--- director.swg.orig	2013-04-19 14:37:44.398037833 +0200
+++ director.swg	2013-04-19 14:38:01.918037522 +0200
@@ -149,7 +149,7 @@ namespace Swig {
     void swig_disconnect_director_self(const char *disconn_method) {
       JNIEnvWrapper jnienv(this) ;
       JNIEnv *jenv = jnienv.getJNIEnv() ;
-      jobject jobj = swig_self_.peek();
+      jobject jobj = swig_self_.get(jenv);
 #if defined(DEBUG_DIRECTOR_OWNED)
       std::cout << "Swig::Director::disconnect_director_self(" << jobj << ")" << std::endl;
 #endif
@@ -161,6 +161,7 @@ namespace Swig {
 #endif
           jenv->CallVoidMethod(jobj, disconn_meth);
         }
+        jenv->DeleteLocalRef(jobj);
       }
     }
 
